python/LAMBDA_DLQ_CHECK/LAMBDA_DLQ_CHECK.py

```diff
@@ -0,0 +1,434 @@
+#
+#        http://aws.amazon.com/apache2.0/
+#
+# or in the "license" file accompanying this file. This file is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for
+# the specific language governing permissions and limitations under the License.
+
+'''
+
+#####################################
+##           Gherkin               ##
+#####################################
+
+Rule Name: LAMBDA_DLQ_CHECK
+
+Description: Checks whether the AWS Lambda function is configured for a Dead Letter Queue(DLQ). The rule is NON_COMPLIANT if the Lambda function is not configured for DLQ. Trigger: Periodic
+
+Trigger:
+  Configuration Change
+
+Reports on:
+  AWS::Lambda::Function
+
+Rule Parameters:
+  dlqArn
+  (Optional) Comma-separated list of SQS and SNS ARN that Lambda function must be configured for as a DLQ
+
+Scenarios:
+  Scenario: 1
+     Given: The dlqArn rule parameter is not valid (not a valid SNS/SQS ARN)
+      Then: Return Error
+  Scenario: 2
+     Given: Lambda function is not configured for DLQ i.e. DeadLetterConfig is not present
+      Then: Return NON_COMPLIANT with Annotation "This Lambda function is not configured for DLQ"
+  Scenario: 3
+     Given: Lambda function is configured for DLQ i.e. DeadLetterConfig is present
+       And: The rule parameter dlqArn is empty
+      Then: Return COMPLIANT
+  Scenario: 4
+      Given: Lambda function is configured for DLQ i.e. DeadLetterConfig is present
+        And: No ARN in the dlqArn rule parameter doesn't match with the DeadLetterConfig ARN
+       Then: Return NON_COMPLIANT with Annotation "This Lambda Function is not associated with the DLQ specified in the "dlqArn" input parameter"
+  Scenario: 5
+      Given: Lambda function is configured for DLQ i.e. DeadLetterConfig is present
+        And: One of the ARN in the dlqArn rule parameter matches with the DeadLetterConfig ARN
+       Then: Return COMPLIANT
+
+'''
+
+import json
+import sys
+import datetime
+import boto3
+import botocore
+
+try:
+    import liblogging
+except ImportError:
+    pass
+
+##############
+# Parameters #
+##############
+
+# Define the default resource to reportpylint to Config Rules
+DEFAULT_RESOURCE_TYPE = 'AWS::Lambda::Function'
+
+# Set to True to get the lambda to assume the Role attached on the Config Service (useful for cross-account).
+ASSUME_ROLE_MODE = False
+
+# Other parameters (no change needed)
+CONFIG_ROLE_TIMEOUT_SECONDS = 900
+
+
+#############
+# Main Code #
+#############
+
+def evaluate_compliance(event, configuration_item, valid_rule_parameters):
+    if 'deadLetterConfig' not in configuration_item['configuration']:
+        return build_evaluation_from_config_item(configuration_item, 'NON_COMPLIANT',
+                                                 annotation='This Lambda function is not configured for DLQ')
+    if 'deadLetterConfig' in configuration_item['configuration']:

```

Comment:
Not necessary since you already check that it is present at line 85 with the negation: `if 'deadLetterConfig' not in configuration_item['configuration']:`